hazelcast/src/main/java/com/hazelcast/query/impl/predicates/OrToInVisitor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl.predicates;
+
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.Indexes;
+import com.hazelcast.util.collection.InternalMultiMap;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.hazelcast.util.collection.ArrayUtils.createCopy;
+
+/**
+ * Transforms predicate (attribute = 1 or attribute = 2 or attribute = 3) into
+ * (attribute in (1, 2, 3)
+ *
+ */
+public class OrToInVisitor extends AbstractVisitor {
```

Comment:
A simple note which explains when/why this transformation is needed would be nice.